src/pages/workspace/tags/WorkspaceTagsPage.tsx

```diff
@@ -89,38 +71,51 @@ function WorkspaceTagsPage({route}: WorkspaceTagsPageProps) {
         setSelectedTags({});
     }, [isFocused]);
 
-    // We currently don't support multi level tags, so let's only get the first level tags.
-    const policyTagLists = useMemo(() => PolicyUtils.getTagLists(policyTags).slice(0, 1), [policyTags]);
-    const tagList = useMemo<PolicyForList[]>(
+    const tagList = useMemo<TagListItem[]>(() => {
+        if (!doesPolicyContainOnlyOneTagList) {
+            return policyTagLists.map((policyTagList) => ({
```

Comment:
This does not look right - above we are checking if `policyTagList.length === 1`, and then in the `else` we're mapping `policyTagList.tags` and flattening that.

I'm pretty confused how this is correct. It seems like we'd want to flatten the tag lists in all cases?

Reply:
The mapping and the flattering process is useless because we have one element in the array, we can just use that element directly. I have pushed changes for that.

Reply:
slack context: https://expensify.slack.com/archives/C036QM0SLJK/p1715121431797619